vnext/Microsoft.ReactNative/JsiApi.idl

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.ReactNative
+{
+  interface IJsiRuntime;
+
+  delegate void JsiDataHandler(UInt8[] data);
+
+  [webhosthidden]
+  interface IJsiBuffer
+  {
+    UInt32 Size { get; };
+    void GetData(JsiDataHandler handler);
+  };
+
+  enum JsiValueType
+  {
+    Undefined,
+    Null,
+    Boolean,
+    Number,
+    Symbol,
+    String,
+    Object,
+  };
+
+  [webhosthidden]
+  struct JsiValueData
+  {
+    JsiValueType Kind;
+    UInt64 Value;
+  };
+
+  [default_interface]
+  runtimeclass JsiPreparedJavaScript
+  {
+  };
+
+  delegate JsiValueData JsiHostFunction(IJsiRuntime runtime, JsiValueData thisValue, JsiValueData[] args);
```

Comment:
>JsiHostFunction [](start = 24, length = 15)

[nit] Given the 'thisValue' parameter, would "...Method" be a better name? If it's meant to also work for functions, wouldn't it be shorter/simpler to accommodate methods by passing 'thisValue' as the first item in the 'args' array?

Reply:
I am following the JSI design here. The 'thisValue' could be undefined if it is not an instance method. It may be difficult to make it part of the args. Anyway, I want to have the code working first end-to-end before trying to change the JSI API design.

---
In reply to: [451845990](https://github.com/microsoft/react-native-windows/pull/5461#discussion_r451845990) [](ancestors = 451845990)